app/my-bets/page.tsx

```diff
@@ -7,96 +7,151 @@ import { Market, Position, tradesOutcomeBalance } from '@/entities';
 
 import { getUserPositions } from '@/queries/conditional-tokens';
 import { UserPosition } from '@/queries/conditional-tokens/types';
-import { getConditionMarket, getMarketUserTrades } from '@/queries/omen';
-import { useQuery, useQueryClient } from '@tanstack/react-query';
-
-import { PropsWithChildren, ReactNode, useCallback, useEffect, useState } from 'react';
+import {
+  Condition,
+  FixedProductMarketMaker,
+  FpmmTrade,
+  getConditionMarket,
+  getMarketUserTrades,
+} from '@/queries/omen';
+import { useQuery, useQueries } from '@tanstack/react-query';
+
+import { PropsWithChildren, ReactNode, useEffect, useMemo, useState } from 'react';
 import { TabBody, TabGroup, TabHeader, TabPanel, TabStyled } from '@swapr/ui';
 import { useAccount } from 'wagmi';
 
+interface UserPositionAll extends UserPosition {
+  fpmmTrades: FpmmTrade[];
+  market: FixedProductMarketMaker;
+  condition: Condition;
+}
+
 export default function MyBetsPage() {
   const { address } = useAccount();
-  const [unredeemedBets, setUnredeemedBets] = useState<UserPosition[]>([]);
-  const queryClient = useQueryClient();
+  const [userPositionsAllData, setUserPositionsAllData] = useState<UserPositionAll[]>([]);
 
-  const { data, isLoading } = useQuery({
+  const { data: userPositionsData, isLoading: isUserPositionsLoading } = useQuery({
     queryKey: ['getUserPositions', address],
     queryFn: () => getUserPositions({ id: address?.toLowerCase() as string }),
     enabled: !!address,
   });
 
-  const userPositions = data?.userPositions ?? [];
+  const userPositions = useMemo(
+    () => userPositionsData?.userPositions ?? [],
+    [userPositionsData]
+  );
+
+  const queries = useMemo(() => {
+    return userPositions.flatMap(userPosition => {
+      const position = new Position(userPosition.position);
+      const outcomeIndex = position.outcomeIndex - 1;
```

Comment:
Why are we substracting 1 from the position outcome index?

Reply:
That's a good question. I don't have a good answer.

This is a detail implementation of the contracts/subgraphs that we're using but I don't think we know the why they did it this way.

@berteotti thoughts?